### ISSUE TITLE
Add a New Loss Term and a Gradient Term which are inconsistent

### ISSUE THREAD

**Issue by ZhichaoWang970201** [pytorch/pytorch#64174]
Question: I want to embed a new loss term into the neural network. However, its gradient is the modification of the derivative of the newly added loss term. I want to know if there is any reference.

The loss term: 
![image](https://user-images.githubusercontent.com/50489235/131341517-ea201c99-19be-41e3-a6be-41cf46f8f197.png)

The original derivative for back propagation:
![image](https://user-images.githubusercontent.com/50489235/131341581-43f3e512-c037-4a31-bce6-fdeae35037a7.png)

The final derivative for back propagation:
![image](https://user-images.githubusercontent.com/50489235/131341646-626bdaf4-15ae-4c14-b018-14532cf8c2fb.png)

How can I add the final derivative into the neural network? Thank you.

**Comment by ezyang** [comment]
Please ask at https://discuss.pytorch.org/
